Pyt 999 deserialization

danilito19 opened the pull request:
Adding triggers for:
pickle.load / loads
yaml.load / load_all

Most of the code is boilerplate and totally relies on @lazorchakp 's awesome setup. Very next PR should be about using some sort of meta-programming or AST re-writing to be able to automate generation of view definitions, url registration, and tests for every framework.

lazorchakp reviewed (changes requested):
This looks really good overall. Very exciting to have some more authors on the repo :)

lazorchakp commented:
I found something else: could you update `base.html` to add deserialization to the side bar on the left hand side? You can check that the UI looks right with `make flask` (or any other framework)

lazorchakp commented on src/vulnpy/flask/blueprint.py:
```diff
@@ -34,6 +38,42 @@ def _cmdi_subprocess_popen():
     return get_template("cmdi.html")
 
 
+@vulnerable_blueprint.route(
```
sorry, now that I looked at the frontend I noticed something else too - can you provide a `/deserialization/` endpoint just like `/cmdi/`? It would just render the `deserialization.html` template and nothing else. Once we can combine the unit tests to be framework-agnostic I think we won't run into cases like this as easily

danilito19 commented:
@lazorchakp great eye! I updated - but I'm still not seeing deserialization in the nav bar even though I updated fragments/base.html .Help? what did I Miss?

lazorchakp commented on src/vulnpy/templates/fragments/base.html:
```diff
@@ -17,6 +17,23 @@
           <div class="main">
             <ul class="menu-list">
               <li><a href="/vulnpy/cmdi/">CMD Injection</a></li>
+              <li><a href="/vulnpy/deserialization/">Untrusted Deserialization</a></li>
```
sorry to keep harping on this. I think the only change we want here is just adding one line for the new list item. There should only be one injection site in the file. The HTML does appear to render properly in the browser, but honestly I'm not sure

lazorchakp reviewed (approved):
looks good!!
